save-core/src/commonMain/kotlin/org/cqfn/save/core/Save.kt

```diff
@@ -73,62 +65,73 @@ class Save(
         // constructing the file path to the configuration file
         val fullPathToConfig = saveProperties.testRootPath!!.toPath() / saveProperties.testConfigPath!!.toPath()
         // get all toml configs in file system
-        val testConfig = ConfigDetector().configFromFile(fullPathToConfig)
-
-        val reporter = getReporter(saveProperties)
-        reporter.beforeAll()
-
-        val plugins: List<Plugin> = discoverPluginsAndUpdateTestConfig(testConfig)
-            .also { it.merge() }
-            // ignoring the section with general configuration as it is not needed to create plugin processors
-            .pluginConfigs.filterNot { it is GeneralConfig }
-            .map {
-                when (it) {
-                    is FixPluginConfig -> FixPlugin(testConfig)
-                    is WarnPluginConfig -> WarnPlugin(testConfig)
-                    else -> throw PluginException("Unknown type <${it::class}> of plugin config was provided")
-                }
+        ConfigDetector()
+            .configFromFile(fullPathToConfig)
+            .getAllTestConfigs()
+            .forEach { testConfig ->
+                val reporter = getReporter(saveProperties)
```

Comment:
I think reporter can be created outside of the loop

Reply:
oh, sure